db/compaction/compaction.cc

```diff
@@ -774,8 +778,18 @@ std::unique_ptr<CompactionFilter> Compaction::CreateCompactionFilter() const {
   CompactionFilter::Context context;
   context.is_full_compaction = is_full_compaction_;
   context.is_manual_compaction = is_manual_compaction_;
+  context.input_start_level = start_level_;
   context.column_family_id = cfd_->GetID();
   context.reason = TableFileCreationReason::kCompaction;
+  context.input_table_properties = GetInputTableProperties();
+  if (context.input_table_properties.empty()) {
+    ROCKS_LOG_WARN(
```

Comment:
I kind of don't want to fail `CreateCompactionFilter()` when `input_table_properties` is empty as I think users can rely on `context.reason` and an empty `input_table_properties` to do some fall back logic to proceed with the compaction filter.

I injected an assert(false) in this to do make check in manual testing. All UT passed.

Reply:
I think eventually FinalizeInputInfo() should detect failures and fail the compaction. Then this would not be reachable

Reply:
Fixed by adding a todo on FinalizeInputInfo()